src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -1378,6 +1403,11 @@ private void RemoveLastInputPrompt()
             private void AddLanguageBuffer()
             {
                 ITextBuffer buffer = _factory.CreateAndActivateBuffer(_window);
+                if (CurrentLanguageBuffer != null)
+                {
+                    _textBufferUndoManagerProvider.RemoveTextBufferUndoManager(CurrentLanguageBuffer);
+                }
+                UndoManager = _textBufferUndoManagerProvider.GetTextBufferUndoManager(buffer);
```

Comment:
Also, how are the undo managers we're wiring up here being connected to the view and undo/redo buttons?